client/events.go

```diff
@@ -46,3 +47,48 @@ func (cli *Client) Events(ctx context.Context, options types.EventsOptions) (io.
 	}
 	return serverResponse.body, nil
 }
+
+// EventsWithHandler decodes the stream of events handing them off to the appropriate
+// handlers for processing. It's up to the caller to stop the processing by canceling
+// the context. The error channel will be closed once the end of the input stream has been
+// reached. If an error is received all processing will be stopped.
+func (cli *Client) EventsWithHandler(ctx context.Context, options types.EventsOptions, handler eventtypes.EventHandler) <-chan error {
```

Comment:
I'd rather see something like this:

```
Events(ctx context.Context, options types.EventsOptions) (<-chan eventtypes.Message, <-chan error)
```

Then, we can have a wrapper built with a handler on top of this.

The issue with the handler approach is that it requires too much boilerplate to get something useful. A handler system can be built on the above, but it is hard to readapt the handler system to channel based system, if that is required instead.